src/main/java/se/knowit/bookitregistration/repository/map/RegistrationRepositoryMapImpl.java

```diff
@@ -1,64 +1,55 @@
-package se.knowit.bookitregistration.service.map;
+package se.knowit.bookitregistration.repository.map;
 
 import se.knowit.bookitregistration.model.Registration;
-import se.knowit.bookitregistration.model.RegistrationValidator;
-import se.knowit.bookitregistration.service.RegistrationService;
+import se.knowit.bookitregistration.repository.RegistrationRepository;
 import se.knowit.bookitregistration.service.exception.ConflictingEntityException;
 
 import java.util.Map;
-import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
-import java.util.stream.Collectors;
 
-public class RegistrationServiceMapImpl implements RegistrationService {
-    private final Map<Long, Registration> registrationStore;
+import static java.util.stream.Collectors.toSet;
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+public class RegistrationRepositoryMapImpl implements RegistrationRepository {
+
     private IdentityHandler identityHandler;
-    private RegistrationValidator registrationValidator;
+    private final Map<Long, Registration> registrationStore;
 
-    public RegistrationServiceMapImpl() {
+    public RegistrationRepositoryMapImpl() {
         this(new ConcurrentHashMap<>());
     }
 
-    RegistrationServiceMapImpl(Map<Long, Registration> registrationStore) {
+    public RegistrationRepositoryMapImpl(Map<Long, Registration> registrationStore) {
         this.registrationStore = registrationStore;
         this.identityHandler = new IdentityHandler(this.registrationStore.size());
-        this.registrationValidator = new RegistrationValidator();
     }
 
     @Override
-    public Set<Registration> findAll() {
-        return Set.copyOf(registrationStore.values());
-    }
-
-    @Override
-    public Registration save(Registration incomingRegistration) throws ConflictingEntityException {
-        Registration validRegistration = registrationValidator.ensureRegistrationIsValidOrThrowException(incomingRegistration);
-        assignRequiredIds(validRegistration);
-        persistRegistration(validRegistration);
-        return validRegistration;
+    public Set<Registration> find(Predicate<Registration> searchFilter) {
+        return registrationStore.values().stream().filter(searchFilter).collect(toSet());
```

Comment:
I think it nice to write every step in the stream as a new line i.e. 

```
return registrationStore.values().stream()
    .filter(searchFilter)
    .collect(toSet());
```

This way it looks more as a pipe ( or a stream going through a number of steps)